### ISSUE TITLE
GET /room_keys/keys?version=$v returns an error if the backup is empty

### ISSUE THREAD

**Issue by manuroe** [matrix-org/synapse#4056]
Discussed at https://matrix.to/#/!uewiilduiDRfPomIha:matrix.org/$15398763772905238xioNB:matrix.org.

Instead returning an empty object, we get this error:
```
{
    errcode = "M_UNKNOWN";
    error = "No room_keys found";
}
```

**Comment by uhoreg** [comment]
It looks like the error is set at https://github.com/matrix-org/synapse/pull/4019/files#diff-5c190c6b2c1bbe2f359c3e41932da2dbR71

I think it makes sense to send an error for `/room_keys/keys/${roomId}/${sessionId}` if no backup is found (though the `errcode = "M_UNKNOWN"` should be fixed), but for `/room_keys/keys/${roomId}` and `/room_keys/keys`, I think it should just send `{"sessions": {}}` or `{"rooms": {}}`.

**Comment by richvdh** [comment]
fixed by #4123
